lib/v8debugapi.js

```diff
@@ -61,29 +61,20 @@ module.exports.create = function(logger_, config_, fileStats_) {
   var breakpoints = {};
   var listeners = {};
   var numBreakpoints = 0;
-  var usePermanentListener = true;
+  // Before V8 4.5, having a debug listener active disables optimization. To
+  // deal with this we only activate the listener when there is a breakpoint
+  // active, and remote it as soon as the snapshot is taken. Furthermore, 4.5
+  // changes the API such that Debug.scripts() crashes unless a listener is
+  // active. We use a permanent listener on V8 4.5+.
+  var usePermanentListener = semver.satisfies(
+    /(\d+\.\d+\.\d+)\.\d+/.exec(process.versions.v8)[1], '>=4.5');
```

Comment:
It would be nice to make this future proof, if V8 ends up using a version doesn't match the 4-dot notation, or a different VM is used. In such a case this code will throw at the `[1]` subscript access